projects/ui-components/lf-selection-list/resize-column.directive.ts

```diff
@@ -0,0 +1,87 @@
+import { Directive, ElementRef, EventEmitter, Input, OnInit, Output, Renderer2 } from '@angular/core';
+import { ColumnDef } from './lf-selection-list-types';
+
+export const COLUMN_MIN_WIDTH: number = 100;
+
+@Directive({
+  selector: '[lfResizeColumn]',
+})
+export class ResizeColumnDirective implements OnInit {
+  @Input('lfResizeColumn') resizable: boolean = false;
+  @Input() columnDef?: ColumnDef;
+  @Output() widthChanged: EventEmitter<number> = new EventEmitter<number>();
+
+  private column: HTMLElement;
+
+  private pressed: boolean = false;
+  resizePosition: number = 0;
+  resizedColumnInitialOffsetLeft: number = 0;
+  viewport?: HTMLElement;
+  th?: HTMLElement;
+  divresizer?: HTMLElement;
+
+  constructor(private renderer: Renderer2, private el: ElementRef) {
+    this.column = this.el.nativeElement;
+  }
+
+  ngOnInit() {
+    if (this.resizable) {
+      this.th = this.renderer.parentNode(this.column);
+      const row = this.renderer.parentNode(this.th);
+      const thead = this.renderer.parentNode(row);
+      const table = this.renderer.parentNode(thead);
+      this.viewport = this.renderer.parentNode(table);
+      this.divresizer = this.renderer.createElement('div');
+      this.divresizer!.classList.add('repository-browser-resize-overlay');
```

Comment:
if not resizable, should we add a css so that the cursor for resize-handle is 'not-allowed' or something that indicates that the column is not resizable?